src/burst_db/build_frame_db.py

```diff
@@ -0,0 +1,743 @@
+# Threshold to use EPSG:3031, Antarctic Polar Stereographic (https://epsg.io/3031)
+SOUTH_THRESHOLD = -60
+SOUTH_EPSG = 3031
+
+
+def make_jpl_burst_id(df: pd.DataFrame):
+    """Make the JPL burst ID from the ESA burst ID."""
+    burst_id_jpl = (
+        "t"
+        + df["relative_orbit_number"].astype(str).str.zfill(3)
+        + "_"
+        + df["burst_id"].astype(str).str.zfill(6)
+        + "_"
+        + df["subswath_name"].str.lower()
+    )
+    return burst_id_jpl
+
+
+def _setup_spatialite_con(con: sqlite3.Connection):
+    """Enable spatialite and load the spatialite extension."""
+    con.enable_load_extension(True)
+    # Try the two versions, mac and linux with .so
+    try:
+        con.load_extension("mod_spatialite")
+    except:
+        con.load_extension("mod_spatialite.so")
+    # Allow us to use spatialite functions in on GPKG files.
+    # https://medium.com/@joelmalone/sqlite3-spatialite-and-geopackages-66a08485da6c
+    con.execute("SELECT EnableGpkgAmphibiousMode();")
+
+
+def make_burst_triplets(df_burst: pd.DataFrame) -> pd.DataFrame:
+    """Make a burst triplets dataframe, aggregating IW1,2,3 from the burst dataframe."""
+
+    def join_track_numbers(orbits: list) -> str:
+        orbits = list(set(orbits))
+        orbits_str = list(map(str, orbits))
+        return ",".join(orbits_str)
+
+    df_burst_triplet_temp = df_burst.dissolve(
+        by="burst_id",
+        aggfunc={
+            "OGC_FID": ["min", "max"],
+            "relative_orbit_number": join_track_numbers,
+            "orbit_pass": "first",
+        },
+        as_index=False,
+    )
+    df_burst_triplet = df_burst_triplet_temp.reset_index(drop=True)
+    df_burst_triplet.columns = [
+        "burst_id",
+        "geom",
+        "OGC_FID_min",
+        "OGC_FID_max",
+        "relative_orbit_numbers",
+        "look_direction",
+    ]
+    return df_burst_triplet
+
+
+def get_land_indicator(gdf: gpd.GeoDataFrame, land_geom: GeometryType.POLYGON):
+    """Get a boolean array indicating if each row of `gdf` intersects `land_geom`."""
+    tree = STRtree(gdf.geometry)
+    idxs_land = tree.query(land_geom, predicate="intersects")
+    if idxs_land.ndim == 2:
+        idxs_land = idxs_land[1]
+    is_in_land = gdf.index.isin(idxs_land)
+    return is_in_land
+
+
+def make_frame_to_burst_table(outfile: str, df_frame_to_burst_id: pd.DataFrame):
+    """Create the frames_bursts table and indexes."""
+    with sqlite3.connect(outfile) as con:
+        _setup_spatialite_con(con)
+
+        df_frame_to_burst_id.to_sql("frames_bursts", con, if_exists="replace")
+        con.execute(
+            "CREATE INDEX IF NOT EXISTS idx_frames_bursts_burst_ogc_fid ON frames_bursts (burst_ogc_fid)"
+        )
+        con.execute(
+            "CREATE INDEX IF NOT EXISTS idx_frames_bursts_frame_fid ON frames_bursts (frame_fid)"
+        )
+        con.execute(
+            "CREATE INDEX IF NOT EXISTS idx_burst_id_map_burst_id ON burst_id_map (burst_id)"
```

Comment:
94 columns...I'll leave it to you decide whether or not to split the line.

Reply:
Letting black split lines